tests/test_streaming.py

```diff
@@ -15,13 +15,14 @@
 @pytest.mark.parametrize('batch_size', [128])
 @pytest.mark.parametrize('drop_last', [False, True])
 @pytest.mark.parametrize('shuffle', [False, True])
-@pytest.mark.parametrize('num_workers', [0, 1, 8])
+@pytest.mark.parametrize('num_workers', [0, 4])
```

Comment:
why change this one? probably still good to test 1 worker?

Reply:
`num_workers` > `0` spins up additional child workers. So `1` means it will spin up 1 child worker and 4 mean 4 child workers. since the underlying behavior is the same (spinning up child workers) so thought of only test with two scenario `0` num_workers and any ( in this case `4`).

Reply:
yeah, i'd like to keep the test around for `1` worker in all cases. I think sometimes 1 is different from >1 in a meaningful way.